starter_code/public/javascript/index.js

```diff
@@ -1,23 +1,105 @@
-const charactersAPI = new APIHandler("http://localhost:8000")
+const charactersAPI = new APIHandler("http://localhost:8000");
 
-$(document).ready( () => {
-  document.getElementById('fetch-all').onclick = function(){
+$(document).ready(() => {
+  document.getElementById("fetch-all").onclick = function() {
+    charactersAPI.getFullList().then(characters => {
+      // document.getElementById("character-info").innerHTML = characters.map(char => char.name);
+      $("#character-container").html("");
+      for (let i = 0; i < characters.length; i++) {
+        $("#character-container").append(`      
+        <div class="character-info" id="character-info">
+        <div class="name">Character Name : ${characters[i].name}</div>
+        <div class="occupation">Character Occupation : ${characters[i].occupation}</div>
+        <div class="debt">Character Debt : ${characters[i].debt}</div>
+        <div class="weapon">Character Weapon : ${characters[i].weapon}</div>
+      </div>
+      `);
+      }
+    });
+  };
 
-  }
-  
-  document.getElementById('fetch-one').onclick = function(){
-    
-  }
-  
-  document.getElementById('delete-one').onclick = function(){
-        
-  }
-  
-  document.getElementById('edit-character-form').onsubmit = function(){
-            
-  }
-  
-  document.getElementById('new-character-form').onsubmit = function(){
-                
-  }
-})
+  document.getElementById("fetch-one").onclick = function() {
+    charactersAPI.getOneRegister($("#search-input").prop("value")).then(character => {
+      // $("#character-container").html(`<pre>${JSON.stringify(characters, null, 2)}</pre>`);
+      $("#character-container").html(`
+        <div class="character-info" id="character-info">
+        <div class="name">Character Name : ${character.name}</div>
+        <div class="occupation">Character Occupation : ${character.occupation}</div>
+        <div class="debt">Character Debt : ${character.debt}</div>
+        <div class="weapon">Character Weapon : ${character.weapon}</div>
+      </div>
+      `);
+    });
+  };
+
+  document.getElementById("delete-one").onclick = function() {
+    charactersAPI.deleteOneRegister($("#delete-input").prop("value"));
+    //.then(character => {
+    // $("#character-container").html(`<pre>${JSON.stringify(characters, null, 2)}</pre>`);
+    //   $("#character-container").html(`
+    //     <div class="character-info" id="character-info">
+    //     <div class="name">Character Name : ${character.name}</div>
+    //     <div class="occupation">Character Occupation : ${character.occupation}</div>
+    //     <div class="debt">Character Debt : ${character.debt}</div>
+    //     <div class="weapon">Character Weapon : ${character.weapon}</div>
+    //   </div>
+    //   `);
+    // });
+  };
+
+  document.getElementById("edit-character-form").onsubmit = function() {
+    event.preventDefault();
```

Comment:
The `event` object is passed as an argument to the callback, so you have to decalre it in the parenthesis of the function:
`document.getElementById("edit-character-form").onsubmit = function(event) { `